plat/arm/common/arm_bl1_setup.c

```diff
@@ -84,17 +85,32 @@ void bl1_early_platform_setup(void)
  *****************************************************************************/
 void arm_bl1_plat_arch_setup(void)
 {
-	arm_setup_page_tables(bl1_tzram_layout.total_base,
-			      bl1_tzram_layout.total_size,
-			      BL_CODE_BASE,
-			      BL1_CODE_END,
-			      BL1_RO_DATA_BASE,
-			      BL1_RO_DATA_END
+#define ARM_MAP_BL1_TOTAL	MAP_REGION_FLAT(			\
+					bl1_tzram_layout.total_base,	\
+					bl1_tzram_layout.total_size,	\
+					MT_MEMORY | MT_RW | MT_SECURE)
+#define ARM_MAP_BL1_CODE	MAP_REGION_FLAT(			\
+					BL_CODE_BASE,			\
+					BL1_CODE_END - BL_CODE_BASE,	\
+					MT_CODE | MT_SECURE)
+#define ARM_MAP_BL1_RO_DATA	MAP_REGION_FLAT(			\
+					BL1_RO_DATA_BASE,		\
+					BL1_RO_DATA_END			\
+						- BL_RO_DATA_BASE,	\
+					MT_RO_DATA | MT_SECURE)
 #if USE_COHERENT_MEM
-			      , BL_COHERENT_RAM_BASE,
-			      BL_COHERENT_RAM_END
+	/* ARM platforms dont use coherent memory in BL1 */
+	assert((BL_COHERENT_RAM_END - BL_COHERENT_RAM_BASE) == 0U);
```

Comment:
I don't think this assert is ever going to be executed. It's only used in the definition of the bl_regions array below and will be ignored by the compiler. I think you should pull this out of the macro definition and use CASSERT instead.

Reply:
I have moved the macros to the top of the file and removed the ARM prefix. I didn't move the definitions into the bl_regions array as for some images we use both regions common to many images (defined in arm_defs.h) and regions specific to the current image, therefore I thought it was clearer to define the macros at the top of the file